src/app/pages/property-details/confirmation-modal/confirmation-modal.component.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { of, throwError } from 'rxjs';
+import {
+  resetSpies,
+  propertyDetailSpy,
+  toastServiceSpy,
+  routerSpy
+} from 'src/app/helpers/spies';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NgxSpinnerModule } from 'ngx-spinner';
+import { ToastrService } from 'ngx-toastr';
+import { ConfirmationModalComponent } from './confirmation-modal.component';
+import { PropertyDetailService } from 'src/app/services/property-detail/property-detail.service';
+import { AppModule } from 'src/app/app.module';
+
+describe('ConfirmationModalComponent', () => {
+  let component: ConfirmationModalComponent;
+  let fixture: ComponentFixture<ConfirmationModalComponent>;
+
+  let mockActivateRouteParam = {
+    paramMap: of({ get: () => 'hello' })
+  };
+
+  beforeAll(() => resetSpies([propertyDetailSpy]));
+  afterEach(() => resetSpies([propertyDetailSpy]));
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ConfirmationModalComponent],
+      imports: [
+        HttpClientTestingModule,
+        RouterTestingModule.withRoutes([
+          { path: '**', component: ConfirmationModalComponent }
+        ]),
+        AppModule,
+        NgxSpinnerModule
+      ],
+      providers: [
+        {
+          provide: PropertyDetailService,
+          useValue: propertyDetailSpy
+        },
+        {
+          provide: ActivatedRoute,
+          useValue: mockActivateRouteParam
+        },
+        {
+          provide: Router,
+          useValue: routerSpy
+        },
+        {
+          provide: ToastrService,
+          useValue: toastServiceSpy
+        }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ConfirmationModalComponent);
+    component = fixture.componentInstance;
+    propertyDetailSpy.deletePropertyService.and.returnValue(
+      of({ message: 'successfully deleted' })
+    );
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should call deleteProperty method', () => {
+    const result = { message: 'successfully deleted' };
+    propertyDetailSpy.deletePropertyService.and.returnValue(
+      of({ message: 'successfully deleted' })
```

Comment:
Hey @MBenedicte , what do you think of having the result returned here so that it can be

```
of(result)
```
Since `result` and the object you have returned are 1 and the same thing.

Reply:
worked on that @Tevinthuku